src/core/jit/model.rs

```diff
@@ -388,6 +420,22 @@ impl<Input> OperationPlan<Input> {
     pub fn size(&self) -> usize {
         self.flat.len()
     }
+
+    pub fn field_is_scalar<Extensions>(&self, field: &Field<Extensions, Input>) -> bool {
+        self.index.type_is_scalar(field.type_of.name())
+    }
+
+    pub fn field_is_enum<Extensions>(&self, field: &Field<Extensions, Input>) -> bool {
+        self.index.type_is_enum(field.type_of.name())
+    }
```

Comment:
Again, the fact that Field is an enum should be known at the time of its creation.

Reply:
we could basically encode this info into field itself, but in that case the field will become another representation of the index. 
While it's lightweighed to store boolean flags in the field, for cases like enum validations we need also to store enum variants that's imply string cloning